src/AutoRest.CSharp/Common/Input/InputTypes/InputOperation.cs

```diff
@@ -10,50 +10,79 @@
 
 namespace AutoRest.CSharp.Common.Input;
 
-internal record InputOperation(
-    string Name,
-    string? ResourceName,
-    string? Summary,
-    string? Deprecated,
-    string Description,
-    string? Accessibility,
-    IReadOnlyList<InputParameter> Parameters,
-    IReadOnlyList<OperationResponse> Responses,
-    RequestMethod HttpMethod,
-    BodyMediaType RequestBodyMediaType,
-    string Uri,
-    string Path,
-    string? ExternalDocsUrl,
-    IReadOnlyList<string>? RequestMediaTypes,
-    bool BufferResponse,
-    OperationLongRunning? LongRunning,
-    OperationPaging? Paging,
-    bool GenerateProtocolMethod,
-    bool GenerateConvenienceMethod,
-    bool KeepClientDefaultValue)
+internal record InputOperation
 {
+    public InputOperation(
+    string name,
+    string? resourceName,
+    string? summary,
+    string? deprecated,
+    string description,
+    string? accessibility,
+    IReadOnlyList<InputParameter> parameters,
+    IReadOnlyList<OperationResponse> responses,
+    RequestMethod httpMethod,
+    BodyMediaType requestBodyMediaType,
+    string uri,
+    string path,
+    string? externalDocsUrl,
+    IReadOnlyList<string>? requestMediaTypes,
+    bool bufferResponse,
+    OperationLongRunning? longRunning,
+    OperationPaging? paging,
+    bool generateProtocolMethod,
+    bool generateConvenienceMethod,
+    bool keepClientDefaultValue,
+    string? operationId = null)
```

Comment:
would like to keep `OperationId` for swagger input, as the user is using it to locate the corresponding operation in swagger.

Reply:
Add an internal `OperationId` property
```csharp
internal string OperationId => ResourceName is null ? Name : $"{ResourceName}_{Name}";
```

Reply:
This is not only used in output models, such as https://github.com/Azure/autorest.csharp/blob/03d3cc83dd33cce6347227622e7a0eb4ec4ac544/src/AutoRest.CSharp/Mgmt/Decorator/OperationExtensions.cs#L58 and https://github.com/Azure/autorest.csharp/blob/03d3cc83dd33cce6347227622e7a0eb4ec4ac544/src/AutoRest.CSharp/Mgmt/Output/Resource.cs#L333.
Would prefer to expose this as a public property instead of constructing it in every single usage.